core/src/main/scala/org/apache/spark/scheduler/local/LocalSchedulerBackend.scala

```diff
@@ -82,9 +88,15 @@ private[spark] class LocalEndpoint(
 
   def reviveOffers() {
     val offers = IndexedSeq(new WorkerOffer(localExecutorId, localExecutorHostname, freeCores))
+    val abortTaskSet = new HashSet[TaskSetManager]()
     for (task <- scheduler.resourceOffers(offers).flatten) {
-      freeCores -= scheduler.CPUS_PER_TASK
-      executor.launchTask(executorBackend, task)
+      val buffer = prepareSerializedTask(scheduler, task,
```

Comment:
Can you add a comment here about why we're serializing (and then immediately deserializing) the task?